lisa/sut_orchestrator/baremetal/bootconfig.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from dataclasses import dataclass, field
+from typing import Optional, Type
+
+from dataclasses_json import dataclass_json
+from marshmallow import validate
+
+from lisa import schema
+from lisa.node import quick_connect
+from lisa.util import InitializableMixin, LisaException, field_metadata, subclasses
+from lisa.util.logger import get_logger
+
+from .schema import BootConfigSchema
+
+
+class BootConfig(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(
+        self,
+        runbook: BootConfigSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.boot_config_runbook: BootConfigSchema = self.runbook
+        self._log = get_logger("boot_config", self.__class__.__name__)
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return BootConfigSchema
+
+    def set_boot_config(self) -> None:
+        raise NotImplementedError()
+
+
+@dataclass_json()
+@dataclass
+class PxeBootSchema(BootConfigSchema):
+    connection: Optional[schema.RemoteNode] = field(
+        default=None, metadata=field_metadata(required=True)
+    )
+    node_setup: str = field(default="", metadata=field_metadata(required=True))
+    node_number: int = field(
+        default=1, metadata=field_metadata(validate=validate.Range(min=1, max=20))
+    )
+    image_source: str = field(default="", metadata=field_metadata(required=True))
+    kernel_append: Optional[str] = field(default="")
+
+
+class PxeBoot(BootConfig):
+    def __init__(
+        self,
+        runbook: PxeBootSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.pxe_runbook: PxeBootSchema = self.runbook
+        self._log = get_logger("pxe_boot", self.__class__.__name__)
+        self._boot_label = "label lisav3"
+        self._boot_dir = "/var/lib/tftpboot/"
+        self._boot_config = self._boot_dir + "pxelinux.cfg/"
```

Comment:
It looks `_boot_dir` and `_boot_config` are used once each. Can it use one variable?